addons/actions/src/lib/types/function/isReserved.js

```diff
@@ -0,0 +1,3 @@
+export default function isReserved(name) {
+  return ['delete'].indexOf(name) >= 0;
```

Comment:
There's a lot of other reserved keywords: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords

You can also consider this approach instead of appending `$`:
https://github.com/rhalff/storybook/blob/master/addons/actions/src/preview.js#L27-L33

Reply:
Tried to solve it by supporting both methods, preferring the one you have suggests.
The eval version could be removed if you find it unnecessary.

I've included the changes of #2428 but moved the check to a different location.